added details on what derivations are, added/removed periods (#1074)
NixOS/nix.dev: source/tutorials/nix-language.md

declaring data and transforming it with functions.

> In practice, describing derivations requires observing the outside world.

Reviewer: ```suggestion
In practice, describing derivations – the Nix language's defining feature, which enables functional programming with the file system – requires observing the outside world.
```

I suggest moving the reminder of what derivations are about to the first sentence, and not repeat it but rather give it another emphasis.

> We discuss [](derivations), how contents of existing files are used to derive new files, further in the tuturial.

Reviewer: ```suggestion
We will discuss [derivations](derivations) later in the tutorial.
```
